### ISSUE TITLE
Spawn simulators and avoid locking

### ISSUE THREAD

**Issue by nirinchev** [realm/realm-js#1251]
The current pipeline locks on simulator when testing react which artificially increases the already long build times. Instead use the recently merged function to spawn a new simulator instance to run the tests: https://github.com/realm/ci/pull/117.

**Comment by blagoev** [comment]
As much as I would like to be done with this it seems .NET and JS are using quite different scheme to run the tests on iOS Simulator.
It seems we are locking the simulator in jenkins script and then a bash scripts builds and runs the actual bundle. To use the provided function we would need to build the package first and then require an external call( the jenkins script) to start the bundle. In addition the test script is also stopping the ROS instance post testing and is used when running tests locally.

So in short it would require significant effort to restructure the test script realm-js uses to support the function in jenkins, which can result is too much churn on the code required for the release.

In my opinion we should move this to later time unless there are some good suggestions on how to avoid too much code changes in the testing structure. @nirinchev what do you think?
